src/native/aws_signing.c

```diff
@@ -204,6 +211,10 @@ static bool s_should_sign_param(const struct aws_byte_cursor *name, void *user_d
 
     (*env)->DeleteLocalRef(env, parameter_name);
 
+    if ((*env)->ExceptionCheck(env)) {
+        return false;
+    }
```

Comment:
don't silently swallow this exception, probably just AWS_FATAL_ASSERT() like we do elsewhere